python/cudf/cudf/_lib/table.pyx

```diff
@@ -0,0 +1,46 @@
+from cudf._lib.cudf cimport *
+
+
+cdef class Table:
+    """
+    Python wrapper around cudf::table
+    """
+    def __cinit__(self, columns=None):
+        cdef gdf_column* c_col
+        if columns is not None:
+            for col in columns:
+                c_col = column_view_from_column(col)
+                self.c_columns.push_back(c_col)
+            self.ptr = new cudf_table(self.c_columns)
+        else:
+            self.ptr = new cudf_table()
```

Comment:
I would if we could use smart pointers here which would allow sharing and reference counting cudf_table objects for example.

Reply:
Since libcudf++ functions will return a `unique_ptr<cudf::table>`, `Table` now wraps a `unique_ptr<cudf::table>`.